ludwig/api.py

```diff
@@ -1898,7 +1898,13 @@ def save_config(self, save_path: str) -> None:
         """
         os.makedirs(save_path, exist_ok=True)
         model_hyperparameters_path = os.path.join(save_path, MODEL_HYPERPARAMETERS_FILE_NAME)
-        save_json(model_hyperparameters_path, self.config_obj.to_dict())
+        save_json(model_hyperparameters_path, self._scrub_creds(self.config_obj).to_dict())
+
+    def _scrub_creds(self, config_obj: ModelConfig) -> ModelConfig:
+        """Returns a copy of the config with all sensitive fields scrubbed."""
+        config_dict = config_obj.to_dict()
+        config_dict.get("backend", {})["credentials"] = {}
```

Comment:
I think we'll have to check if `credentials` is in the backend config. The reason is that the `backend` is the only part of a Ludwig config that hasn't been schemified yet. Otherwise it'll complain about an access/type error if the `credentials` key ins't in the backend config